runtime/decimal.c

```diff
@@ -54,7 +54,10 @@ TaggedPtrPanicCode finish(decQuad *dq, decContext *cx) {
     }
     else {
         result.panicCode = 0;
-        if (decQuadClass(dq) == DEC_CLASS_NEG_ZERO) {
```

Comment:
You are callling decQuadClass 4 times!

Also why you are testing for DEC_CLASS_POS_ZERO?

I don't think you can test this until you have done mul or div. So please add those with some tests.

Reply:
> You are callling decQuadClass 4 times!
> 
Fixed

> Also why you are testing for DEC_CLASS_POS_ZERO?
> 
For a case like this `-1E-6143 + 1E-6143`, expected value in subset is `0`. But if we did not handle like this we are getting  `0E-6143`. That is why I checked +Zero case too, and added a test for that

Reply:
I will add mul into this PR.